benchmarks/cpp/nvfuser/dynamic_shape.cpp

```diff
@@ -0,0 +1,177 @@
+  auto aten_output = std::get<0>(aten_results);
+  auto aten_mean = std::get<1>(aten_results);
+  auto aten_rstd = std::get<2>(aten_results);
+
+  fec = std::make_unique<FusionExecutorCache>(std::move(fusion_ptr));
+  aten_inputs = {
+      aten_grad_out, aten_input, aten_mean, aten_rstd, aten_weight, aten_bias};
+  auto cg_outputs = fec->runFusionWithInputs(aten_inputs);
+
+  return fec->getMostRecentKernelRuntime();
+}
+
+static void LayerNormBackward_HeuristicLookup(
+    benchmark::State& benchmark_state) {
+  std::unique_ptr<Fusion> fusion_ptr = std::make_unique<Fusion>();
+  FusionGuard fg(fusion_ptr.get());
+
+  // PreAllocate
+  std::unique_ptr<FusionExecutorCache> fec;
+  std::vector<at::IValue> aten_inputs;
+
+  std::vector<int64_t> shape{20, 100, 35, 67};
+  std::vector<int64_t> norm_shape{67};
+
+  auto runtime = getLayerBackwardNormRuntime(
+      std::move(fusion_ptr), fec, aten_inputs, shape, norm_shape);
+  TORCH_INTERNAL_ASSERT(
+      runtime->getMaybeHeuristicsFor(aten_inputs).has_value());
+
+  for (auto _ : benchmark_state) {
+    // Setup (not included in the measurement)
+    runtime->getMaybeHeuristicsFor(aten_inputs);
+  }
+}
+
+static auto getLayerForwardNormRuntime(
+    std::unique_ptr<Fusion> fusion_ptr,
+    std::unique_ptr<FusionExecutorCache>& fec,
+    std::vector<at::IValue>& aten_inputs,
+    std::vector<int64_t>& shape,
+    std::vector<int64_t>& norm_shape) {
+  Fusion& fusion = *fusion_ptr.get();
+
+  const float kEps = 1e-5;
+  Double* eps_ptr = new Double(kEps);
+
+  auto input = makeSymbolicTensor(shape.size());
+  fusion.addInput(input);
+
+  auto result = layer_norm(input, norm_shape, nullptr, nullptr, eps_ptr);
+
+  fusion.addOutput(result.output);
+  fusion.addOutput(result.mean);
+  fusion.addOutput(result.invstd);
+
+  auto options = at::TensorOptions().dtype(at::kFloat).device(at::kCUDA, 0);
+  at::Tensor aten_input = at::randn(shape, options);
+
+  fec = std::make_unique<FusionExecutorCache>(std::move(fusion_ptr));
+  aten_inputs = {aten_input};
+  auto cg_outputs = fec->runFusionWithInputs(aten_inputs);
+
+  return fec->getMostRecentKernelRuntime();
+}
+
+static void LayerNormForward_HeuristicLookup(
+    benchmark::State& benchmark_state) {
+  std::unique_ptr<Fusion> fusion_ptr = std::make_unique<Fusion>();
+  FusionGuard fg(fusion_ptr.get());
+
+  // PreAllocate
+  std::unique_ptr<FusionExecutorCache> fec;
+  std::vector<at::IValue> aten_inputs;
+
+  std::vector<int64_t> shape{20, 100, 35, 67};
+  std::vector<int64_t> norm_shape{67};
+
+  auto runtime = getLayerForwardNormRuntime(
+      std::move(fusion_ptr), fec, aten_inputs, shape, norm_shape);
+  TORCH_INTERNAL_ASSERT(
+      runtime->getMaybeHeuristicsFor(aten_inputs).has_value());
+
+  for (auto _ : benchmark_state) {
+    // Setup (not included in the measurement)
+    runtime->getMaybeHeuristicsFor(aten_inputs);
+  }
+}
+
+BENCHMARK(LayerNormBackward_HeuristicLookup)->Unit(benchmark::kMicrosecond);
```

Comment:
Maybe rename this file to `heuristic.cpp`. It seems you are measuring the heuristic lookup time. Also, the shapes are static.